[test]:improve coverage for ext_handler in edge/pkg/metamanager/metaserver

Dhiren-Mhatre opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, please read our contributor guidelines: https://github.com/kubeedge/kubeedge/blob/master/CONTRIBUTING.md
2. Ensure you have added or ran the appropriate tests for your PR
-->
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
This PR adds comprehensive test coverage for the handlerfactory package, increasing the code coverage from 50% to 84%. 

**Which issue(s) this PR fixes**:
Part of Enhance KubeEdge testing coverage #6186


**Does this PR introduce a user-facing change?**:

NONE

wbc6080 commented on edge/pkg/metamanager/metaserver/handlerfactory/ext_handler_test.go:
```diff
@@ -302,3 +310,213 @@ func TestExec(t *testing.T) {
+		handler.ServeHTTP(w, req)
+
+		resp := w.Result()
+		body, _ := io.ReadAll(resp.Body)
```
Add the returned err processing, and the other similarities in this PR need also be added

wbc6080 reviewed (commented):
Overall looks good

wbc6080 commented:
[edge-node-tasks-status-enhancement.md](https://github.com/kubeedge/kubeedge/pull/6243/files#diff-30334b49bf910455441d57f789045af47f0e5f24de70699a93fc569d67ac2f3a) This file is irrelevant with this PR. It seems that you made a mistake when rebasing. @Dhiren-Mhatre

Dhiren-Mhatre commented:
> [edge-node-tasks-status-enhancement.md](https://github.com/kubeedge/kubeedge/pull/6243/files#diff-30334b49bf910455441d57f789045af47f0e5f24de70699a93fc569d67ac2f3a) This file is irrelevant with this PR. It seems that you made a mistake when rebasing. @Dhiren-Mhatre

Thanks for pointing that out! I've corrected the issue and removed the irrelevant file from the PR.

wbc6080 commented:
/lgtm

WillardHu commented:
/approve
